woocommerce/woocommerce
# CYS - E2E tests: add homepage picker E2E tests

> [!WARNING]  
> This PR is blocked by #45937 


### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Part of https://github.com/woocommerce/woocommerce/issues/45344


This PR fixes a bug that does not add the focus on the pattern after the user clicks on "Design your homepage".

Also, it adds E2E tests for the homepage picker. Futhermore, I noticed that the `getEditor` util was wrong, and I fixed it (in the page, there are multiple iframe with that class). For this reason, it was necessary update snapshots as well.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that the CI is green.
2. Head over to WooCommerce > Home > Customize your store
3. Click on "Start designing" and proceed to the Pattern Assembler.
4. Click on "Design your homepage".
5. Ensure that the first pattern is highlighted.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 
CYS - E2E tests: add homepage picker E2E tests

</details>
